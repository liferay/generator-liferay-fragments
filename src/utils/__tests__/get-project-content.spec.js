const getTestFixtures = require('../get-test-fixtures');
const {
  default: getProjectContent,
} = require('../project-content/get-project-content');

describe('utils/get-project-content', () => {
  it('requires a valid package.json file', () => {
    expect(() => getProjectContent(Math.random().toString())).toThrow();
  });

  getTestFixtures().forEach((projectPath) => {
    it('matches a project structure', () => {
      const projectContent = getProjectContent(projectPath);

      expect(projectContent).toEqual(
        expect.objectContaining({
          basePath: projectPath,

          project: expect.objectContaining({
            name: expect.any(String),
          }),

          collections: expect.arrayContaining([
            expect.objectContaining({
              slug: expect.any(String),
              fragmentCollectionId: expect.any(String),

              metadata: expect.objectContaining({
                name: expect.any(String),
                description: expect.any(String),
              }),

              fragments: expect.arrayContaining([
                expect.objectContaining({
                  slug: expect.any(String),
                  html: expect.any(String),
                  js: expect.any(String),
                  css: expect.any(String),

                  metadata: expect.objectContaining({
                    htmlPath: expect.any(String),
                    jsPath: expect.any(String),
                    cssPath: expect.any(String),
                    name: expect.any(String),
                  }),
                }),
              ]),
            }),
          ]),
        })
      );
    });
  });
});
