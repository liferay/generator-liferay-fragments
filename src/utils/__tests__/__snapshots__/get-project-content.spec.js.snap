// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/get-project-content matches a project structure 1`] = `
Object {
  "basePath": "",
  "collections": Array [
    Object {
      "fragmentCollectionId": "sample-collection",
      "fragmentCompositions": Array [],
      "fragments": Array [
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration.json",
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample component fragment name",
            "type": "component",
          },
          "slug": "sample-component-fragment",
        },
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration.json",
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample fragment name",
          },
          "slug": "sample-fragment",
        },
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration.json",
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample section fragment name",
            "type": "section",
          },
          "slug": "sample-section-fragment",
        },
      ],
      "metadata": Object {
        "description": "Sample collection description",
        "name": "Sample collection name",
      },
      "slug": "sample-collection",
    },
  ],
  "pageTemplates": Array [],
  "project": Object {
    "name": "sample-project-content",
  },
}
`;

exports[`utils/get-project-content matches a project structure 2`] = `
Object {
  "basePath": "",
  "collections": Array [
    Object {
      "fragmentCollectionId": "sample-collection",
      "fragmentCompositions": Array [],
      "fragments": Array [
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration/configuration.json",
            "cssPath": "c/s/s/sample-fragment.css",
            "htmlPath": "h/t/m/l/sample-fragment.html",
            "jsPath": "j/s/sample-fragment.js",
            "name": "Sample fragment name",
          },
          "slug": "sample-fragment",
        },
      ],
      "metadata": Object {
        "description": "Sample collection description",
        "name": "Sample collection name",
      },
      "slug": "sample-collection",
    },
  ],
  "pageTemplates": Array [],
  "project": Object {
    "name": "sample-project-with-custom-paths",
  },
}
`;

exports[`utils/get-project-content matches a project structure 3`] = `
Object {
  "basePath": "",
  "collections": Array [
    Object {
      "fragmentCollectionId": "sample-collection",
      "fragmentCompositions": Array [],
      "fragments": Array [
        Object {
          "configuration": "{
  \\"fieldSets\\": [
    {
      \\"fields\\": [
      ]
    }
  ]
}
",
          "css": ".sample-fragment {
  color: inherit;
  font-size: 18px;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI',
    'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
    'Helvetica Neue', sans-serif;
}
",
          "html": "<div class=\\"sample-fragment\\">
  Sample fragment
</div>
",
          "js": "console.group('Sample fragment');
console.log('fragmentElement', fragmentElement);
console.groupEnd();
",
          "metadata": Object {
            "configurationPath": "configuration.json",
            "cssPath": "styles.css",
            "htmlPath": "index.html",
            "jsPath": "main.js",
            "name": "Sample fragment with thumbnail",
            "thumbnailPath": "thumbnail.png",
            "type": "section",
          },
          "slug": "sample-fragment",
        },
      ],
      "metadata": Object {
        "description": "Sample collection autogenerated",
        "name": "Sample collection",
      },
      "slug": "sample-collection",
    },
  ],
  "pageTemplates": Array [],
  "project": Object {
    "description": "Liferay Fragments project",
    "engines": Object {
      "node": ">= 8.0.0",
      "npm": ">= 6.0.0",
    },
    "keywords": Array [
      "liferay",
      "liferay-fragments",
    ],
    "name": "sample-liferay-fragments",
    "scripts": Object {
      "add-collection": "yo liferay-fragments:collection",
      "add-fragment": "yo liferay-fragments:fragment",
      "add-fragment-composition": "yo liferay-fragments:fragment-composition",
      "add-page-template": "yo liferay-fragments:page-template",
      "compress": "yo liferay-fragments:compress",
      "export": "yo liferay-fragments:export",
      "import": "yo liferay-fragments:import",
      "import:watch": "yo liferay-fragments:import --watch",
      "preview": "yo liferay-fragments:preview",
    },
    "version": "1.0.0",
  },
}
`;

exports[`utils/get-project-content matches a project structure 4`] = `
Object {
  "basePath": "",
  "collections": Array [
    Object {
      "fragmentCollectionId": "sample-collection",
      "fragmentCompositions": Array [],
      "fragments": Array [
        Object {
          "configuration": "",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample fragment name",
          },
          "slug": "sample-fragment",
        },
      ],
      "metadata": Object {
        "description": "Sample collection description",
        "name": "Sample collection name",
      },
      "slug": "sample-collection",
    },
  ],
  "pageTemplates": Array [],
  "project": Object {
    "name": "sample-project-without-fragment-configuration",
  },
}
`;
